game-core/src/test/java/games/strategy/engine/framework/startup/ui/posted/game/pbf/ForumPosterEditorViewModelTest.java

```diff
@@ -129,45 +131,49 @@ void viewForumPostNoOpOnInvalidData(final String forumSelection, final String to
   }
 
   private ForumPosterEditorViewModel givenViewModelWithInvalidFieldSettings(
-      final String forumSelection, final String topicId) {
+      final String topicId, final String username, final char[] password) {
     final ForumPosterEditorViewModel viewModel = new ForumPosterEditorViewModel(() -> {});
-    viewModel.setForumSelection(forumSelection);
     viewModel.setTopicId(topicId);
-    Preconditions.checkState(
-        !viewModel.isTopicIdValid(), "Test makes sense only if topic id is invalid");
+    viewModel.setForumUsername(username);
+    viewModel.setForumPassword(password);
     return viewModel;
   }
 
   @SuppressWarnings("unused")
   private static List<Arguments> invalidForumSettings() {
     return List.of(
-        Arguments.of("forumSelection", "not a number"),
-        Arguments.of("forumSelection", ""),
-        Arguments.of("forumSelection", "0.0"),
-        Arguments.of("forumSelection", "0"),
-        Arguments.of("forumSelection", "-1"));
+        // invalid topic IDs
+        Arguments.of("not a number", "user", new char[] {'a'}),
+        Arguments.of("", "user", new char[] {'a'}),
+        Arguments.of("0.0", "user", new char[] {'a'}),
+        Arguments.of("0", "user", new char[] {'a'}),
+        Arguments.of("-1", "user", new char[] {'a'}),
+        // missing user
+        Arguments.of("1", "", new char[] {'a'}),
+        // missing password
+        Arguments.of("1", "", new char[] {}));
```

Comment:
```suggestion
        Arguments.of("1", "", new char[0]));
```

But in this scenario I would be ok with keeping the initializer curly braces because of consistency